Explain more clearly how to use rest/db/file (#817)
syncthing/docs: rest/db-file-get.rst

GET /rest/db/file
=================

Returns data about a given file, including version and
availability.

Request
-------

> The call requires parameters for `folder` and `file` in the querystring:

Reviewer: ```suggestion
The call requires parameters for `folder` and `file` in the query string:
```

> - `folder` is the folder code which you can find in the synthing UI, e.g. `5camp-slpa8`

Reviewer: ```suggestion
- `folder` is the folder ID which you can find in the Synthing Web GUI, e.g. `5camp-slpa8`
```

> - `folder` is the folder code which you can find in the synthing UI, e.g. `5camp-slpa8`

Reviewer 2: It's called ID in the UI and other docs, so I'd use that here too.

```suggestion
- `folder` is the folder ID which you can find in the synthing UI, e.g. `5camp-slpa8`
```

Author: applied, thanks!

> - `folder` is the folder ID which you can find in the Synthing Web GUI, e.g. `5camp-slpa8`

Reviewer: ```suggestion
- `folder` is the folder ID which you can find in the Syncthing Web GUI, e.g. `5camp-slpa8`
```

Missed the typo.

Author: Fix applied, thanks!

> - `file` is the full path to the file you are interested in. The path and filename must be correctly url-encoded.

Reviewer: ```suggestion
- `file` is the full path starting from the folder root to the file you are interested in. The path and filename must be correctly URL-encoded.
```

> - `file` is the full path to the file you are interested in. The path and filename must be correctly url-encoded.

Reviewer 2: To me "full path" indicates an absolute file path, while the path here is relative to the folder root path. And seems confusing to mention "path and filename", as if they were somehow separate, I'd just say `The path must ...`.

Author: Good point, alternate wording added in 89ee8d1ab11b5bb1fe9fcf5b3daac1211d9af55e

Example Request
^^^^^^^^^^^^^^^

>   curl --silent --get --header "X-API-KEY: YOUR_API_TOKEN" "http://localhost:8080/rest/db/file?folder=YOUR_FOLDER_ID" --data-urlencode "file=path/to/file.pdf"

Reviewer: ```suggestion
  curl --silent --get --header "X-API-KEY: YOUR_API_TOKEN" "http://localhost:8384/rest/db/file?folder=YOUR_FOLDER_ID" --data-urlencode "file=path/to/file.pdf"
```

GET /rest/db/file
=================

Returns data about a given file, including version and availability.

Request
-------

The call requires parameters for `folder` and `file` in the query string:

- `folder` is the folder ID which you can find in the Syncthing Web GUI,
  e.g. `5camp-slpa8`.

- `file` is the relative path of the file starting from the folder root to the
  file you are interested in.  The path and filename must be correctly
  URL-encoded.

Example Request
^^^^^^^^^^^^^^^

.. code-block:: bash

>     curl --silent --get --header "X-API-KEY: YOUR_API_TOKEN" "http://localhost:8384/rest/db/file?folder=YOUR_FOLDER_ID" --data-urlencode "file=path/to/file.pdf"

Reviewer 3: There are already numerous examples involving `curl` in the docs.  I'd find it convenient to adopt the same CLI options used there, e.g. `-H` instead of `--header` and omit the `--get`.

      "local": { /* a file entry */ }
    }

> ``local`` and ``global`` refer to the current file on disk and the globally newest file, respectively.

Reviewer: ```suggestion
``local`` and ``global`` refer to the current file on the local device and the globally newest file, respectively.
```

      "local": { /* a file entry */ }
    }

``local`` and ``global`` refer to the current file on the local device and the

> globally newest file, respectively.  A file entry looks like this:

Reviewer 3: There is a factual error here.  The `mtime` attribute is not part of the respective file entry, but top-level next to `availability`.  When that is fixed, the outer object wrapping can go away in this code sample.
